datafusion/functions/src/datetime/now.rs

```diff
@@ -77,12 +77,13 @@ impl ScalarUDFImpl for NowFunc {
         &self.signature
     }
 
-    fn return_field_from_args(&self, _args: ReturnFieldArgs) -> Result<Field> {
+    fn return_field_from_args(&self, _args: ReturnFieldArgs) -> Result<FieldRef> {
         Ok(Field::new(
```

Comment:
So in theory we could update this code to create the `FieldRef` once on creation, and then return an `Arc::clone` rather than re-creating the Field each time -- perhaps we can do that as some follow on PRs.